src/textual/widget.py

```diff
@@ -948,6 +965,9 @@ def _to_widget(child: int | Widget, called: str) -> Widget:
             cast("int | Widget", before if after is None else after), "move towards"
```

Comment:
The overloads above are smart enough to detect issues when calling the function with none of `after`/`before` or with both but they're not smart enough to infer that one of the two is not `None` inside the function body so I couldn't get rid of this `cast`.